nums/core/array/blockarray.py

```diff
@@ -445,6 +445,11 @@ def tensordot(self, other, axes=2):
         if not isinstance(other, BlockArray):
             raise ValueError("Cannot automatically construct BlockArray for tensor operations.")
 
+        if not isinstance(axes, int):
+            raise TypeError(f"'{type(axes).__name__}' object cannot be interpreted as an integer")
```

Comment:
The `tensordot` supports non-integer axes; could be a tuple or array like structure of integers. Can we update this check as follows:

1. If it's an integer, ok.
2. If it's "array like", raise `NotImplementedError`. You can use `nums.core.array.utils.is_array_like`.
3. If it's neither an int or array like, then raise a type error, indicating that an integer or array like was expected, but got type(axes) instead.

Reply:
For the type error, can we update this to say f"Unexpected axes type '{type(axes).__name__}'"